unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarModifiers/AvatarModifierArea.cs

```diff
@@ -61,19 +61,58 @@ public override IEnumerator ApplyChanges(string newJson)
 
     private void OnDestroy()
     {
-        Destroy(collider);
+        RemoveAllModifiers();
+    }
+
+    private void Update()
+    {
+        if (model?.area == null)
+        {
+            return;
+        }
+
+        // Find avatars currently on the area
+        HashSet<GameObject> newAvatarsInArea = new HashSet<GameObject>(DetectAllAvatarsInArea());
+
+        // Call event for avatars that just entered the area
+        foreach (GameObject avatarThatEntered in newAvatarsInArea.Except(avatarsInArea))
+        {
+            OnAvatarEnter?.Invoke(avatarThatEntered);
+        }
+
+        // Call events for avatars that just exited the area
+        foreach (GameObject avatarThatExited in avatarsInArea.Except(newAvatarsInArea))
+        {
+            OnAvatarExit?.Invoke(avatarThatExited);
+        }
```

Comment:
If there is no change in the avatars contained inside the hashset, we should escape before traversing all the elements, to avoid that unnecessary performance cost